docs/src/custom.md

```diff
@@ -1,3 +1,232 @@
 # Defining custom `LinearMap` types
 
-TBD
\ No newline at end of file
+In this section, we want to demonstrate on a simple, actually built-in, linear map type
+how to define custom `LinearMap` subtypes. First of all, `LinearMap{T}` is an extendable
+abstract type, where `T` denotes the `eltype`.
+
+## Basics
+
+As an example, we want to define a map type whose objects correspond to lazy analogues
+of `fill`ed matrices. Naturally, we need to store the filled value `λ` and the `size`
+of the linear map.
+
+```julia
+using LinearMaps, LinearAlgebra
+
+struct MyFillMap{T} <: LinearMaps.LinearMap{T}
+    λ::T
+    size::Dims{2}
+    function MyFillMap(λ::T, dims::Dims{2}) where {T}
+        all(≥(0), dims) || throw(ArgumentError("dims of MyFillMap must be non-negative"))
+        promote_type(T, typeof(λ)) == T || throw(InexactError())
+        return new{T}(λ, dims)
+    end
+end
+```
+
+By default, for any `A::MyFillMap{T}`, `eltype(A)` returns `T`. Upon application to a
+vector `x` and/or interaction with other `LinearMap` objects, we need to check consistent
+sizes.
+
+```julia
+Base.size(A::MyFillMap) = A.size
+```
+
+By a couple of defaults provided for all subtypes of `LinearMap`, we only need to define
+a `LinearAlgebra.mul!` method to have minimal, operational type.
+
+```julia
+function LinearAlgebra.mul!(y::AbstractVecOrMat, A::MyFillMap, x::AbstractVector)
+    LinearMaps.check_dim_mul(y, A, x)
+    return fill!(y, iszero(A.λ) ? zero(eltype(y)) : A.λ*sum(x))
+end
+```
+
+Again, due to generic fallbacks the following now "just work":
+
+* out-of-place multiplication `A*x`,
```

Comment:
```suggest
* out-of-place multiplication by a vector on left or right: `A*x` and `y'*A`

Reply:
Well, within the flow of the help page it doesn't work yet, only after dealing with the transpose/adjoint further below. I'll add a comment there.